### PR TITLE
Merge changes from v0.36.0-rc0 to master

### PR DESCRIPTION
**Pull request by astralcai**

### CODE REVIEW

**Comment by mlxd** [comment]
Can we create a separate branch for merging changes into master, rather than trying to update the 0.36.0-rc0 branch?

**Review by mlxd — CHANGES_REQUESTED** [review]
Blocking on this as it would be better to create a separate branch than to modify the RC for a master merge.

**Comment by astralcai** [comment]
> Blocking on this as it would be better to create a separate branch than to modify the RC for a master merge.

Should I simply target this at master then?

File: tests/drawer/test_draw_catalyst.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 import pennylane as qml
 
+pytest.skip(allow_module_level=True)
```
Comment (astralcai) [inline tests/drawer/test_draw_catalyst.py:20]: Should I unskip this now?

Comment (mlxd) [reply tests/drawer/test_draw_catalyst.py:20]: Yep, that can be removed. With the catalyst addition, all should go green

File: tests/test_compiler.py [hunk 2]
```diff
@@ -32,6 +32,7 @@
 from jax.core import ShapedArray  # pylint:disable=wrong-import-order, wrong-import-position
 
 # pylint: disable=too-few-public-methods, too-many-public-methods
+pytest.skip(allow_module_level=True)
```
Comment (astralcai) [inline tests/test_compiler.py:35]: And unskip this as well?
```suggestion

```

File: tests/drawer/test_draw_catalyst.py (hunk 1 above)
Comment (astralcai) [reply tests/drawer/test_draw_catalyst.py:20]:
```suggestion

```

File: tests/test_compiler.py (hunk 2 above)
Comment (mlxd) [reply tests/test_compiler.py:35]: Yep, that too.

**Pull request closed without merging**